static/app/views/starfish/views/spans/spanDetails.tsx

```diff
@@ -0,0 +1,35 @@
+import {DataRow} from 'sentry/views/starfish/modules/databaseModule/databaseTableView';
+import QueryDetail from 'sentry/views/starfish/modules/databaseModule/panel';
+import EndpointDetail, {
+  EndpointDataRow,
+} from 'sentry/views/starfish/views/endpointDetails';
+import {SpanDataRow} from 'sentry/views/starfish/views/spans/spansTable';
+
+type SpanDetailBodyProps = {
+  row: SpanDataRow;
+};
+export default function SpanDetail({
+  row,
+  onClose,
+}: Partial<SpanDetailBodyProps> & {onClose: () => void}) {
+  // Types need to be fixed here
+  switch (row?.span_operation) {
+    case 'http.client':
+      return <EndpointDetail row={row as unknown as EndpointDataRow} onClose={onClose} />;
+    case 'db':
+      return (
+        <QueryDetail
+          row={row as unknown as DataRow}
```

Comment:
I can work on updating this, but I think now that we are making this component more generic with this PR, we should update the props to represent what the component needs vs what will be passed into it. This will eliminate the need to do this castings.
For example in `QueryDetail` instead of the `row` prop being of type `DataRow`, it would be of type
`{description: string; p50: string, ...}` including everything it needs from data row but nothing more.